File: 80. Remove Duplicates from Sorted Array II/remove.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashMap<Integer,Integer> map=new HashMap<>();
        ArrayList<Integer> al=new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
            
        }
        int t=0;
        for(int i=0;i<nums.length;i++){
            int freq=map.get(nums[i]);
            if((freq>=2)&&(!(al.contains(nums[i])))){
                al.add(nums[i]);
                al.add(nums[i]);
            }
            else if((freq<2)&&(!(al.contains(nums[i]))))
            al.add(nums[i]);
        }
        System.out.println(al);
        for(int i=0;i<al.size();i++)
        nums[i]=al.get(i);
        return(al.size());
    }
}

File: 82. Remove Duplicates from Sorted List II/remove.cpp
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        HashMap<Integer,Integer> map=new HashMap<>();
        ListNode copy=head;
        if(copy == null)
        return copy;
        while(copy!=null){
            int val=copy.val;
            map.put(val,map.getOrDefault(val,0)+1);
            copy=copy.next;
        }
        ArrayList<Integer> al=new ArrayList<>();
        for(Map.Entry<Integer,Integer> entry : map.entrySet()){
        int key=entry.getKey();
        if(map.get(key)==1){
        System.out.println("value pushed -> "+key);
        al.add(key);
        }
        }
        Collections.sort(al);
        copy=head;
        for(int i=0;i<al.size();i++){
            copy.val=al.get(i);
            if(i!=al.size()-1)
            copy=copy.next;
        }
        if(al.size()>=1){
        copy.next=null;
        return head;
        }
        else return null;
    }
}
